### PR TITLE
review/jm-initial

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
This PR is intended to be a review of the initial build and should not be merged. Please see comments below to be addressed and applied to the `main` branch. Please respond to the comments once the updates are applied or the comment addressed. The comments will then be resolved following re-review.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,7 +6,7 @@
 # 📣 What does this dbt package do?
 - Produces modeled tables that leverage Reddit Ads data from [Fivetran's connector](https://fivetran.com/docs/applications/reddit-ads) in the format described by [this ERD](https://fivetran.com/docs/applications/reddit-ads#schemainformation) and builds off the output of our [Reddit Ads source package](https://github.com/fivetran/dbt_reddit_ads_source).
 - Enables you to better understand the performance of your ads across varying grains:
-  - Providing an account, campaign, ad group, keyword, ad, utm and search level reports.
+  - Providing an account, campaign, ad group, keyword, ad, utm and search level reports.
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:9]: There is no keyword report. We should probably remove this from the package details.

File: DECISIONLOG.md [hunk 2]
```diff
@@ -1 +1,2 @@
 # Decision Log 
+
```
Comment (fivetran-joemarkiewicz) [inline DECISIONLOG.md:2]: Is there a need for a decisionlog in this package?

File: README.md [hunk 3]
```diff
@@ -136,8 +136,8 @@ packages:
...
 
-## Opinionated Decisions
-In creating this package, which is meant for a wide range of use cases, we had to take opinionated stances on a few different questions we came across during development. We've consolidated significant choices we made in the [DECISIONLOG.md](https://github.com/fivetran/dbt_reddit_ads/blob/main/DECISIONLOG.md), and will continue to update as the package evolves. We are always open to and encourage feedback on these choices, and the package in general.
+## Opinionated Decisions 
+In creating this package, which is meant for a wide range of use cases, we had to take opinionated stances on a few different questions we came across during development. We've consolidated significant choices we made in the [DECISIONLOG.md](https://github.com/fivetran/dbt_reddit_ads/blob/main/DECISIONLOG.md), and will continue to update as the package evolves. We are always open to and encourage feedback on these choices, and the package in general.
```
Comment (fivetran-joemarkiewicz) [inline README.md:140]: Same comment as before, is there a need for the decisionlog in this package?

Comment (fivetran-joemarkiewicz) [reply README.md:140]: It looks like we may want to include a blurb about the default behavior of the url model. See how we phrased this for [twitter_ads](https://github.com/fivetran/dbt_twitter/blob/main/DECISIONLOG.md)

File: README.md [hunk 4]
```diff
@@ -14,7 +14,7 @@
 # 📣 What does this dbt package do?
 - Produces modeled tables that leverage Reddit Ads data from [Fivetran's connector](https://fivetran.com/docs/applications/reddit-ads) in the format described by [this ERD](https://fivetran.com/docs/applications/reddit-ads#schemainformation) and builds off the output of our [Reddit Ads source package](https://github.com/fivetran/dbt_reddit_ads_source).
 - Enables you to better understand the performance of your ads across varying grains:
-  - Providing an account, campaign, ad group, keyword, ad, utm and search level reports.
+  - Providing an account, campaign, ad group, keyword, ad, utm and search level reports.
```
Comment (fivetran-joemarkiewicz) [inline README.md:17]: Same comment, we should remove the keyword reference.

File: README.md [hunk 5]
```diff
@@ -98,7 +98,7 @@ models:
 ### Change the source table references
 If an individual source table has a different name than the package expects, add the table name as it appears in your destination to the respective variable:
 
-> IMPORTANT: See this project's [`dbt_project.yml`](https://github.com/fivetran/dbt_reddit_ads/blob/main/dbt_project.yml) variable declarations to see the expected names.
+> IMPORTANT: See this project's [`dbt_project.yml`](https://github.com/fivetran/dbt_reddit_ads/blob/main/dbt_project.yml) variable declarations to see the expected names.
```
Comment (fivetran-joemarkiewicz) [inline README.md:101]: This should reference the source package in order to highlight the sources used. The transform only shows the staging models referenced.

File: models/reddit_ads.yml [hunk 6]
```diff
@@ -15,10 +15,10 @@ models:
...
             - time_zone_id
     columns:
-      - name: date_day
-        description: '{{ doc("date_day") }}'
-      - name: account_id
-        description: '{{ doc("account_id") }}'
+      - name: date_day 
+        description: '{{ doc("date_day") }}' 
+      - name: account_id 
+        description: '{{ doc("account_id") }}'
```
Comment (fivetran-joemarkiewicz) [inline models/reddit_ads.yml:21]: Should we apply not_null tests to these fields?

File: models/reddit_ads.yml [hunk 7]
```diff
@@ -47,14 +47,14 @@ models:
             - campaign_id
             - currency
     columns:
-      - name: date_day
-        description:  '{{ doc("date_day") }}'
+      - name: date_day 
+        description:  '{{ doc("date_day") }}'
```
Comment (fivetran-joemarkiewicz) [inline models/reddit_ads.yml:51]: Should we apply not_null tests to these fields?

File: models/reddit_ads.yml [hunk 8]
```diff
@@ -47,14 +47,14 @@ models:
...
+      - name: date_day 
+        description:  '{{ doc("date_day") }}' 
       - name: account_id
         description:  '{{ doc("account_id") }}'
       - name: ad_group_name
         description:  '{{ doc("ad_group_name") }}'
-      - name: ad_group_id
-        description:  '{{ doc("ad_group_id") }}'
+      - name: ad_group_id 
+        description:  '{{ doc("ad_group_id") }}'
```
Comment (fivetran-joemarkiewicz) [inline models/reddit_ads.yml:57]: Should we apply not_null tests to these fields?

File: models/reddit_ads.yml [hunk 9]
```diff
@@ -83,10 +83,10 @@ models:
...
             - currency
     columns:
-      - name: date_day
-        description: '{{ doc("date_day") }}'
-      - name: ad_id
-        description: '{{ doc("ad_id") }}'
+      - name: date_day 
+        description: '{{ doc("date_day") }}' 
+      - name: ad_id 
+        description: '{{ doc("ad_id") }}'
```
Comment (fivetran-joemarkiewicz) [inline models/reddit_ads.yml:89]: Should we apply not_null tests to these fields?

File: models/reddit_ads.yml [hunk 10]
```diff
@@ -122,14 +122,14 @@ models:
             - campaign_id
             - currency
     columns:
-      - name: date_day
-        description: '{{ doc("date_day") }}'
+      - name: date_day 
+        description: '{{ doc("date_day") }}'
```
Comment (fivetran-joemarkiewicz) [inline models/reddit_ads.yml:126]: Should we apply not_null tests to these fields?

File: models/reddit_ads.yml [hunk 11]
```diff
@@ -122,14 +122,14 @@ models:
...
+      - name: date_day 
+        description: '{{ doc("date_day") }}' 
       - name: account_id
         description: '{{ doc("account_id") }}'
       - name: campaign_name
         description: '{{ doc("campaign_name") }}'
-      - name: campaign_id
-        description: '{{ doc("campaign_id") }}'
+      - name: campaign_id 
+        description: '{{ doc("campaign_id") }}'
```
Comment (fivetran-joemarkiewicz) [inline models/reddit_ads.yml:132]: Should we apply not_null tests to these fields?

File: models/reddit_ads.yml [hunk 12]
```diff
@@ -154,12 +154,12 @@ models:
             - post_id
             - currency
     columns:
-      - name: date_day
-        description: '{{ doc("date_day") }}'
+      - name: date_day 
+        description: '{{ doc("date_day") }}'
```
Comment (fivetran-joemarkiewicz) [inline models/reddit_ads.yml:158]: Should we apply not_null tests to these fields?

File: models/reddit_ads.yml [hunk 13]
```diff
@@ -154,12 +154,12 @@ models:
...
-      - name: date_day
-        description: '{{ doc("date_day") }}'
+      - name: date_day 
+        description: '{{ doc("date_day") }}' 
       - name: ad_name
         description: '{{ doc("ad_name") }}'
-      - name: ad_id
-        description: '{{ doc("ad_id") }}'
+      - name: ad_id 
+        description: '{{ doc("ad_id") }}'
```
Comment (fivetran-joemarkiewicz) [inline models/reddit_ads.yml:162]: Should we apply not_null tests to these fields?

File: models/reddit_ads__account_report.sql [hunk 14]
```diff
@@ -26,6 +26,7 @@ accounts as (
         sum(report.spend) as spend
 
         {{ fivetran_utils.persist_pass_through_columns(pass_through_variable='reddit_ads__account_passthrough_metrics', transform = 'sum') }}
+
```
Comment (fivetran-joemarkiewicz) [inline models/reddit_ads__account_report.sql:29]: To be consistent with our style standards, would you mind providing an additional space before the `from` following the macro call. Could you then apply the same to the remaining models.

File: models/reddit_ads__url_report.sql [hunk 15]
```diff
@@ -57,6 +57,8 @@ joined as (
         sum(report.impressions) as impressions,
         sum(report.spend) as spend
 
+        
+
```
Comment (fivetran-joemarkiewicz) [inline models/reddit_ads__url_report.sql:61]: Can we add the `{{ fivetran_utils.persist_pass_through_columns(pass_through_variable='reddit_ads__ad_passthrough_metrics', transform = 'sum') }}` macro to this model.

**Review by fivetran-joemarkiewicz — COMMENTED** [review]
@fivetran-reneeli great job on this package! I have just a few minor suggestions and questions. Please respond and make the corresponding updates in the `main` branch. Let me know if you have any questions!

File: CHANGELOG.md (hunk 1 above)
Comment (fivetran-reneeli) [reply CHANGELOG.md:9]: thanks for catching, updated

File: DECISIONLOG.md (hunk 2 above)
Comment (fivetran-reneeli) [reply DECISIONLOG.md:2]: not yet! removed

File: README.md (hunk 3 above)
Comment (fivetran-reneeli) [reply README.md:140]: Do we need it? Reddit ads URLs are pretty straightforward from my understanding, with one url per ad. The twitter blurb is regarding how there can be multiple links per tweet, so requires an index

File: models/reddit_ads.yml (hunk 6 above)
Comment (fivetran-reneeli) [reply models/reddit_ads.yml:21]: true, updated

File: models/reddit_ads.yml (hunk 7 above)
Comment (fivetran-reneeli) [reply models/reddit_ads.yml:51]: updated

File: models/reddit_ads.yml (hunk 8 above)
Comment (fivetran-reneeli) [reply models/reddit_ads.yml:57]: updated

File: models/reddit_ads.yml (hunk 9 above)
Comment (fivetran-reneeli) [reply models/reddit_ads.yml:89]: updated

File: models/reddit_ads.yml (hunk 10 above)
Comment (fivetran-reneeli) [reply models/reddit_ads.yml:126]: updated

File: models/reddit_ads.yml (hunk 12 above)
Comment (fivetran-reneeli) [reply models/reddit_ads.yml:158]: updated

File: models/reddit_ads.yml (hunk 13 above)
Comment (fivetran-reneeli) [reply models/reddit_ads.yml:162]: updaetd

File: models/reddit_ads__account_report.sql (hunk 14 above)
Comment (fivetran-reneeli) [reply models/reddit_ads__account_report.sql:29]: done!

File: models/reddit_ads__url_report.sql (hunk 15 above)
Comment (fivetran-reneeli) [reply models/reddit_ads__url_report.sql:61]: oops, forgot to add here-- added

File: README.md (hunk 3 above)
Comment (fivetran-joemarkiewicz) [reply README.md:140]: Makes sense! Let's remove.

**Comment by fivetran-joemarkiewicz** [comment]
This looks great! Thanks for making all the suggested updates @fivetran-reneeli.

Closing out this PR 🎉

**Pull request closed without merging**